### ISSUE TITLE
Error when using keyboard to change date

### ISSUE THREAD

**Issue by sameervk** [wojtekmaj/react-date-picker#387]
When I am trying to use a key (up, down, number) to change the value of the date, it is giving an error, although it works nicely in your demo. I am using the code exactly as in your example. 

import React, { useState } from 'react';
import DatePicker from 'react-date-picker';

function MyApp() {
  const [value, onChange] = useState(new Date());

  return (
    <div>
      <DatePicker
        onChange={onChange}
        value={value}
      />
    </div>
  );
}

I am new to react, so I am not sure where I am going wrong. Thank you. Please see below attached screenshot of the error.

![image](https://user-images.githubusercontent.com/16178000/135884022-287b3e5a-dcf4-44ee-ab7a-b342b33ea460.png)

**Comment by clamborg** [comment]
Here's a simple codepen https://codesandbox.io/s/small-framework-5uqgj?file=/src/App.js

Hitting this as well - Many thanks!

**Comment by nguyennick197** [comment]
I was also hitting this issue and it seemed to be breaking regardless of the date picker version so I had a guess that an update to one of its dependencies was the source. I checked which dependencies were updated recently and figured it might be the "make-event-props" package. Adding
```
"resolutions": {
    "react-date-picker/make-event-props": "1.2.0"
},
```
to my package.json fixed the issue for me at least.

**Comment by clamborg** [comment]
> I was also hitting this issue and it seemed to be breaking regardless of the date picker version so I had a guess that an update to one of its dependencies was the source. I checked which dependencies were updated recently and figured it might be the "make-event-props" package. Adding
> 
> ```
> "resolutions": {
>     "react-date-picker/make-event-props": "1.2.0"
> },
> ```
> 
> to my package.json fixed the issue for me at least.

Thanks @nguyennick197 great work around for now - much appreciated!

**Comment by savetheclocktower** [comment]
This one is biting an app I work on; our carelessness in not pinning to a specific version of this library means that we inherited this bug from `react-date-picker`'s recent update. @wojtekmaj, if the `make-event-props` package is indeed the cause, then you could fix the underlying issue in that repo instead. But since I haven't verified that on my own, I'm just chiming in to confirm that this appears to be a pretty big regression.

**Comment by savetheclocktower** [comment]
For other readers of this issue, I should also note that @nguyennick197's proposed workaround is great if you use Yarn, but there's not yet an equivalent workaround for NPM, short of using [npm-force-resolutions](https://github.com/rogeriochaves/npm-force-resolutions).

**Comment by wojtekmaj** [comment]
Fixed in d6e30877e89d4f40ae57d26d85fa11f69520512d. Sorry!

**Comment by clamborg** [comment]
thanks for the quick turnaround @wojtekmaj - confirmed working for us with patch!
